### PR TITLE
document densities from sprint 1 nll refactor

### PR DESCRIPTION
**Pull request by Andrea-Havron-NOAA**
# What is the feature?

*  Addresses issues #612 

# How have you implemented the solution?

* Added documentation of class members and functions. Also added documentation on the top of each file that explained the acronyms LPDF and LPMF.

# Does the PR impact any other area of the project?

# How to test this change

doxygen gha will pass
# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: inst/include/distributions/functors/multinomial_lpmf.hpp [hunk 1]
```diff
@@ -13,20 +26,28 @@ namespace fims_distributions
...
-        std::vector<bool> is_na;
+        Type nll = 0.0; /*!< total negative log-likelihood contribution of the distribution */
+        fims::Vector<size_t> dims; /*!< Dimensions of the number of rows and columns of the multivariate dataset */
+        std::vector<bool> is_na; /*!< Boolean; if true, data observation is NA and the likelihood contribution for the entire row is skipped */
         #ifdef TMB_MODEL
-        ::objective_function<Type> *of;
+        ::objective_function<Type> *of; /*!< Pointer to the TMB objective function */
         #endif
-        // data_indicator<tmbutils::vector<Type> , Type> keep;
+        // data_indicator<tmbutils::vector<Type> , Type> keep;  /*!< total negative log-likelihood contribution of the distribution */
```
Comment (cmlegault) [inline inst/include/distributions/functors/multinomial_lpmf.hpp:35]: I think this comment should be /*!< Indicator used in TMB one-step-ahead residual calculations */

File: inst/include/distributions/functors/normal_lpdf.hpp [hunk 2]
```diff
@@ -11,22 +24,30 @@ namespace fims_distributions {
...
+    fims::Vector<Type> mu; /*!< mean of the distribution; can be a vector or scalar */
+    fims::Vector<Type> sd; /*!< standard deviation of the distribution; can be a vector or scalar */
+    Type nll = 0.0; /*!< total negative log-likelihood contribution of the distribution */
+    std::vector<bool> is_na; /*!< Boolean; if true, data observation is NA and the likelihood contribution is skipped */
     #ifdef TMB_MODEL
-    ::objective_function<Type> *of;
+    ::objective_function<Type> *of; /*!< Pointer to the TMB objective function */
     #endif
-    //data_indicator<tmbutils::vector<Type> , Type> keep;
+    //data_indicator<tmbutils::vector<Type> , Type> keep; /*!< total negative log-likelihood contribution of the distribution */
```
Comment (cmlegault) [inline inst/include/distributions/functors/normal_lpdf.hpp:35]: I think this comment should be /*!< Indicator used in TMB one-step-ahead residual calculations */

File: inst/include/distributions/functors/lognormal_lpdf.hpp [hunk 3]
```diff
@@ -13,22 +25,30 @@ namespace fims_distributions
...
         {
         }
 
+        /** @brief Destructor.
+         */
         virtual ~LogNormalLPDF() {}
 
+        /**
+         * @brief Evaluates the negative log-likelihood of the lognormal probability density function
+         * @param do_log Boolean; if true, log densities are returned
```
Comment (cmlegault) [inline inst/include/distributions/functors/lognormal_lpdf.hpp:50]: If I am being overly pedantic, please ignore this comment. Are log densities actually returned? Or are log densities used in the calculation of the negative log likelihood, which is returned? Apologies if I am mis-interpreting the intent of this documentation. This same comment applies to the other functions as well, but is not repeated in those other places.

Comment (Andrea-Havron-NOAA) [reply inst/include/distributions/functors/lognormal_lpdf.hpp:50]: I have a note to refactor the do_log argument in sprint 2 so it is fixed at TRUE given how we apply these functions in FIMS

**Comment by KyleShertzer-NOAA** [comment]
The new documentation is an improvement. Looks good to me.

**Pull request merged**